From a55ece179817756d1b0445ce3b42e6a91978f3ab Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sat, 4 Jul 2026 16:43:36 +0200
Subject: [PATCH 1/1] Use cinnamon schema for idle-delay

Signed-off-by: c4pp4
---
 panels/screen/cc-screen-panel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/screen/cc-screen-panel.c b/panels/screen/cc-screen-panel.c
index 1f62986..a1c56d7 100644
--- a/panels/screen/cc-screen-panel.c
+++ b/panels/screen/cc-screen-panel.c
@@ -496,7 +496,7 @@ cc_screen_panel_init (CcScreenPanel *self)
                     G_CALLBACK (on_lock_settings_changed),
                     self);
   self->priv->gsd_settings = g_settings_new ("com.canonical.unity.settings-daemon.plugins.power");
-  self->priv->session_settings = g_settings_new ("org.gnome.desktop.session");
+  self->priv->session_settings = g_settings_new ("org.cinnamon.desktop.session");
   self->priv->lockdown_settings = g_settings_new ("org.gnome.desktop.lockdown");
   g_signal_connect (self->priv->lockdown_settings,
                     "changed",
-- 
2.54.0

