From b5cb2eeb03c3c21ca17b1277fdc733e276ad6e64 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sun, 24 Oct 2021 16:30:36 +0200
Subject: [PATCH 1/1] Use ubuntu help when in Unity

Signed-off-by: c4pp4
---
 src/screenshot-utils.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/screenshot-utils.c b/src/screenshot-utils.c
index fa342d8..5260de8 100644
--- a/src/screenshot-utils.c
+++ b/src/screenshot-utils.c
@@ -137,10 +137,17 @@ screenshot_display_help (GtkWindow *parent)
 {
   g_autoptr(GError) error = NULL;
 
+  if (in_desktop("Unity")) {
+    gtk_show_uri_on_window (parent,
+                            "help:ubuntu-help/screen-shot-record",
+                            gtk_get_current_event_time (),
+                            &error);
+  } else {
   gtk_show_uri_on_window (parent,
                           "help:gnome-help/screen-shot-record",
                           gtk_get_current_event_time (),
                           &error);
+  }
 
   if (error)
     screenshot_show_dialog (parent,
-- 
2.35.1

