From 66c23e9980c799af74cec1f42ccebde35bf292e4 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sat, 13 Jan 2024 01:14:16 +0100
Subject: [PATCH 1/1] Fix the first run functionality (LP #1435033)

Signed-off-by: c4pp4
---
 libdiodon/zeitgeist-clipboard-storage.vala | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/libdiodon/zeitgeist-clipboard-storage.vala b/libdiodon/zeitgeist-clipboard-storage.vala
index 485eeac..a44f296 100644
--- a/libdiodon/zeitgeist-clipboard-storage.vala
+++ b/libdiodon/zeitgeist-clipboard-storage.vala
@@ -66,12 +66,10 @@ namespace Diodon
 
         public ZeitgeistClipboardStorage()
         {
-            File file = File.new_for_path (Utils.get_database_file_path());
-
             // disables local Zeitgeist database reader to avoid race condition
             // when database is initially created.
             // only do this if database file is not already created
-            if(!Utils.using_in_memory_database() && !file.query_exists()) {
+            if(!Utils.using_in_memory_database()) {
                 debug("Setting default value of ZEITGEIST_LOG_DIRECT_READ to 0");
                 Environment.set_variable("ZEITGEIST_LOG_DIRECT_READ", "0", false);
             }
-- 
2.41.0

