diff -Nur a/debian/guest-account.sh b/debian/guest-account.sh
--- a/debian/guest-account.sh	2017-04-04 23:26:00.000000000 +0200
+++ b/debian/guest-account.sh	2017-08-04 04:03:40.131691067 +0200
@@ -152,6 +152,15 @@
     source_local_prefs
   fi
 
+  # set possible local guest session GSettings defaults
+  if [ -f ${site_gs}/gsettings/99_default.gschema.override ]; then
+    local x
+    for x in ${site_gs}/gsettings/*; do
+      cat "${x}" | sed '/^\[.*\]$/ s/\./\//g' | \
+        su - ${GUEST_USER} -c 'dbus-launch dconf load /'
+    done
+  fi
+
   echo ${GUEST_USER}
 }
 
