From 2c2339a4358dee940f07361efde6ae4c4c071b45 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sun, 10 Aug 2025 22:24:34 +0200
Subject: [PATCH 1/1] Disable global menu via UBUNTU_MENUPROXY

Signed-off-by: c4pp4
---
 gtk/gtkapplication-dbus.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gtk/gtkapplication-dbus.c b/gtk/gtkapplication-dbus.c
index 18f2d8e..c55c1f4 100644
--- a/gtk/gtkapplication-dbus.c
+++ b/gtk/gtkapplication-dbus.c
@@ -506,7 +506,8 @@ gtk_application_impl_dbus_startup (GtkApplicationImpl *impl,
       g_value_unset (&value);
     }
 
-  if (!same_bus)
+  const char *menu_proxy = g_getenv ("UBUNTU_MENUPROXY");
+  if (!same_bus || (menu_proxy && g_strcmp0 (menu_proxy, "0") == 0))
     {
       g_object_set (gtk_settings_get_default (),
                     "gtk-shell-shows-app-menu", FALSE,
-- 
2.49.1

