From 738f4ed9153ff6930e667867397343a55cf26112 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Fri, 1 May 2026 23:03:23 +0200
Subject: [PATCH 1/1] Use GTK3 and libappindicator

Signed-off-by: c4pp4
---
 CMakeLists.txt               | 1 +
 cmake/FindAPPINDICATOR.cmake | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d70078b..b31e42d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -311,6 +311,7 @@ endif()
 if(ENABLE_GTK)
     tr_get_required_flag(ENABLE_GTK GTK_IS_REQUIRED)
 
+    set(USE_GTK_VERSION 3)
     if(USE_GTK_VERSION STREQUAL "AUTO" OR USE_GTK_VERSION EQUAL 4)
         pkg_check_modules(GTK4
             gtkmm-4.0>=${GTKMM_MINIMUM}
diff --git a/cmake/FindAPPINDICATOR.cmake b/cmake/FindAPPINDICATOR.cmake
index c398b13..94f6758 100644
--- a/cmake/FindAPPINDICATOR.cmake
+++ b/cmake/FindAPPINDICATOR.cmake
@@ -10,7 +10,7 @@ find_library(AYATANA_APPINDICATOR_LIBRARY
         ayatana-appindicator
     HINTS ${PC_AYATANA_APPINDICATOR_LIBRARY_DIRS})
 
-if(AYATANA_APPINDICATOR_INCLUDE_DIR AND AYATANA_APPINDICATOR_LIBRARY)
+if(NOT (AYATANA_APPINDICATOR_INCLUDE_DIR AND AYATANA_APPINDICATOR_LIBRARY))
     set(APPINDICATOR_INCLUDE_DIR ${AYATANA_APPINDICATOR_INCLUDE_DIR})
     set(APPINDICATOR_LIBRARY ${AYATANA_APPINDICATOR_LIBRARY})
     set(APPINDICATOR_IS_AYATANA ON)
-- 
2.53.0

