diff '--color=auto' -Nur a/configure.ac b/configure.ac
--- a/configure.ac	2025-07-11 16:23:26.000000000 +0200
+++ b/configure.ac	2025-08-25 14:22:54.163613175 +0200
@@ -1778,6 +1778,7 @@
     AS_HELP_STRING([--disable-gtk3],
         [Determines whether to use Gtk+ 3.0 vclplug on platforms where Gtk+ 3.0 is available.]),
 ,test "${test_gtk3}" = no -o "${enable_gtk3+set}" = set || enable_gtk3=yes)
+enable_gtk3=yes
 
 AC_ARG_ENABLE(gtk4,
     AS_HELP_STRING([--enable-gtk4],
@@ -1829,6 +1830,7 @@
     AS_HELP_STRING([--disable-gio],
         [Determines whether to use the GIO support.]),
 ,test "${enable_gio+set}" = set || enable_gio=yes)
+enable_gio=yes
 
 AC_ARG_ENABLE(qt5,
     AS_HELP_STRING([--enable-qt5],
@@ -2152,6 +2154,7 @@
     AS_HELP_STRING([--disable-dconf],
         [Disable the dconf configuration backend (enabled by default where
          available).]))
+enable_dconf=yes
 
 libo_FUZZ_ARG_ENABLE(formula-logger,
     AS_HELP_STRING(
